kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java

```diff
@@ -130,10 +132,13 @@ public Snapshot buildLatestSnapshot(Engine engine)
     public Snapshot getSnapshotAt(
             Engine engine,
             long version) throws TableNotFoundException {
+        SnapshotImpl upperBoundSnapshot = getSnapshotAtInit(engine);
```

Comment:
This regresses the current path where the table is not a commit coordinated table with the extra snapshot loading. Discussed offline with @dhruvarya-db. Lets figure out a way to avoid the regression.

Reply:
Added two tests for the new way of snapshot loading for the `getSnapshotAt` API.